tests/test_encoding.rs

```diff
@@ -0,0 +1,153 @@
+                match MLSCiphertext::decode(&mut Cursor::new(&encrypted_message_bytes)) {
+                    Ok(a) => a,
+                    Err(err) => panic!("Error decoding MLSCiphertext: {:?}", err),
+                };
+            assert_eq!(encrypted_message, encrypted_message_decoded);
+        }
+        // Test encoding/decoding of Proposal messages
+
+        // Updates
+
+        // TODO: Leaving out Extensions completely still yields a decoding
+        // error! Should be solved by #164.
+        let capabilities_extension = Box::new(CapabilitiesExtension::default());
+        let lifetime_extension = Box::new(LifetimeExtension::new(60));
+        let mandatory_extensions: Vec<Box<dyn Extension>> =
+            vec![capabilities_extension, lifetime_extension];
+
+        let key_package_bundle = KeyPackageBundle::new(
+            &[group_state.ciphersuite().name()],
+            credential_bundle,
+            mandatory_extensions,
+        )
+        .unwrap();
+        let update = group_state.create_update_proposal(
+            &[],
+            credential_bundle,
+            key_package_bundle.get_key_package().clone(),
+        );
+        let update_encoded = update.encode_detached().unwrap();
+        let update_decoded = match MLSPlaintext::decode(&mut Cursor::new(&update_encoded)) {
+            Ok(a) => a,
+            Err(err) => panic!("Error decoding MPLSPlaintext Update: {:?}", err),
+        };
+
+        assert_eq!(update, update_decoded);
+
+        // Adds
+        let capabilities_extension = Box::new(CapabilitiesExtension::default());
+        let lifetime_extension = Box::new(LifetimeExtension::new(60));
+        let mandatory_extensions: Vec<Box<dyn Extension>> =
+            vec![capabilities_extension, lifetime_extension];
+
+        let key_package_bundle = KeyPackageBundle::new(
+            &[group_state.ciphersuite().name()],
+            credential_bundle,
+            mandatory_extensions,
+        )
+        .unwrap();
+        let add = group_state.create_add_proposal(
+            &[],
+            credential_bundle,
+            key_package_bundle.get_key_package().clone(),
+        );
+        let add_encoded = add.encode_detached().unwrap();
+        let add_decoded = match MLSPlaintext::decode(&mut Cursor::new(&add_encoded)) {
+            Ok(a) => a,
+            Err(err) => panic!("Error decoding MPLSPlaintext Add: {:?}", err),
+        };
+
+        assert_eq!(add, add_decoded);
+
+        // Removes
+        let remove =
+            group_state.create_remove_proposal(&[], credential_bundle, LeafIndex::from(1u32));
+        let remove_encoded = remove.encode_detached().unwrap();
+        let remove_decoded = match MLSPlaintext::decode(&mut Cursor::new(&remove_encoded)) {
+            Ok(a) => a,
+            Err(err) => panic!("Error decoding MPLSPlaintext Remove: {:?}", err),
+        };
+
+        assert_eq!(remove, remove_decoded);
+
+        // Commits
```

Comment:
All these different messages should go into their own tests. Move the setup above into a separate function and add individual test functions for the different tests.